frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,66 @@
+import { getRelevantTemplates } from '../../../../../selectors/vm-template/selectors';
+import { getFieldId } from '../../../utils/vm-settings-tab-utils';
+import { UpdateOptions } from '../../types';
+import { iGetVmSettingAttribute } from '../../../selectors/immutable/vm-settings';
+import { iGetLoadedCommonData } from '../../../selectors/immutable/selectors';
+import { VMSettingsField, VMWizardProps, VMSettingsRenderableField } from '../../../types';
+import { CommonTemplatesValidation } from './validation-types';
+import { getValidationsFromTemplates, getFieldValidations } from './selectors';
+
+// TODO: Add all the fields in the form
+// For each field we need to check for validations
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[] => {
+  // Proceed if there are no validation for the specific fieldId
+  if (!(fieldId in IDToJsonPath)) {
+    return [];
+  }
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  // Get templates based on OS, WorkloadProfile and Flavor
+  const relevantTemplates = getRelevantTemplates(commonTemplates, os, workloadProfile, flavor);
+
+  // Get all the validations from the relevant templates
+  const validations = getValidationsFromTemplates(relevantTemplates);
+
+  // Get all the validations which are relevant for the field
+  const fieldValidations = getFieldValidations(validations, IDToJsonPath[fieldId]);
+
+  // Check for each validation if it is a 'between' validation
+  fieldValidations.forEach((v) => {
+    v.integerBetween = 'min' in v && 'max' in v;
+    v.stringBetween = 'minLength' in v && 'maxLength' in v;
+  });
+
+  return fieldValidations;
+};
+
+export const runValidation = (validation: CommonTemplatesValidation, value: number): boolean => {
```

Comment:
can we do ?
```js
export const runValidations = (validations: CommonTemplatesValidation[], value: number): (valid: boolean, mesage: string) => {
   let message = null
    const valid = validations.some((validation) => {
          message = validation.message
           ...
            if ('max' in validation) {
                 return value <= validation.max;
            }
           return false
      }
      ...
      return { valid, messge}
  });
```

Reply:
sorry the signiture should be like in:
see: https://github.com/openshift/console/pull/3839/files#r362222830

``` js
const runVlidations = (validations, value: any) => ...
```
because this method can accept any type of falue number of string and run the appropriate validation rules.